.github/scripts/parse_ported_tests.sh

```diff
@@ -12,8 +12,20 @@ WORKSPACE_PATH="${2:-$GITHUB_WORKSPACE}"
 
 echo "Changed or new test files: $CHANGED_TEST_FILES"
 
-# Extract ported_from markers
-uv run fill $CHANGED_TEST_FILES --show-ported-from --clean --quiet --links-as-filled --skip-coverage-missed-reason --ported-from-output-file ported_from_files.txt
+FILTERED_FILES=""
+for file in $CHANGED_TEST_FILES; do
+    if git diff origin/main -- "$file" | grep -q "^+.*@pytest.mark.ported_from"; then
```

Comment:
I feel we could use the [.github/scripts/fill_prepatched_tests.sh](https://github.com/ethereum/execution-spec-tests/blob/main/.github/scripts/fill_prepatched_tests.sh#L19-L22) trick where we checkout the main branch, do  ` uv run fill $CHANGED_TEST_FILES --show-ported-from --clean --quiet --links-as-filled --skip-coverage-missed-reason --ported-from-output-file ported_from_files.txt` both in `main` and the branch with the changes, diff the two `ported_from_files.txt` files, and only consider new ported tests the ones that only appear in the new branch.